content/blog/prototyping/index.md

```diff
@@ -0,0 +1,726 @@
+    Note {
+        title: String::from("Draft"),
+        content: String::from("My first note")
+    }
+}
+```
+
+If you have a type that you need to move between threads (i.e. it needs to be `Send`), you can use `Arc<Mutex<T>>` to get around the borrow checker.
+If you're worried about performance, remember that other languages like Python or Java do this implicitly behind your back.
+
+```rust
+use std::sync::{Arc, Mutex};
+use std::thread;
+
+let note = Arc::new(Mutex::new(Note {
+    title: String::from("Draft"),
+    content: String::from("My first note")
+}));
+
+let note_clone = Arc::clone(&note);
+thread::spawn(move || {
+    let mut note = note_clone.lock().unwrap();
+    note.content.push_str(" with additions");
+});
+```
+
+([Playground](https://play.rust-lang.org/?version=stable&mode=debug&edition=2021&gist=4b93a53ebc1d7ee6bc2b39c91543fba7))
+
+### Use A Flat Module Hierarchy
+
+Your `main.rs` is a great place for the entire prototype. 
+That's fine because you can always split it up later as soon as this becomes "uncomfortable."
+By then, you have a clear vision of what things should be called and how they should interact.
+
+I keep modules in the same file until they grow too large:
+
+```rust
+mod foo {
+    pub fn bar() {
+        println!("Hello from foo::bar");
+    }
+}
+
+mod baz {
+    pub fn qux() {
+        println!("Hello from baz::qux");
+    }
+}
+
+fn main() {
+    foo::bar();
+    baz::qux();
+}
+```
+
+This allows me to quickly move around in the codebase (as it's just one file) and also move code around by just cutting and pasting from one scope to another.
+It's very cheap.
+The individual modules can be moved to their own files later.
+
+On top of that, the individual modules can be treated like a black box.
+When I'm working on a library, this helps me to quickly test the API and see if the naming makes sense and if the modules are structured correctly.
+E.g. I might have a `config` module with a `load` function that returns a `Config` struct, which gets used by the `server` module to start the server. 
+
+```rust
+mod config {
+    pub struct Config {
+        pub port: u16,
+    }
+
+    pub fn load() -> Config {
+        Config { port: 8080 }
+    }
+}
+
+mod server {
+    use crate::config;
+
+    pub struct Server {
+        config: config::Config,
+    }
+
+    impl Server {
+        pub fn new(config: config::Config) -> Self {
+            Server { config }
+        }
+
+        pub fn start(&self) {
+            println!("Starting server on port {}", self.config.port);
+        }
+    }
+}
+
+use server::Server;
+
+fn main() {
+    let config = config::load();
+    let server = Server::new(config);
+    server.start();
+}
+```
+
+In general, I try to keep a flat hierarchy in the beginning and only introduce more structure when it becomes necessary.
+See also [Matklad's article on large Rust workspaces](https://matklad.github.io/2021/08/22/large-rust-workspaces.html).
+
+## Summary
+
+That's it!
```

Comment:
You make it sound like it's 3 simple tricks :joy: 

Maybe make an actual summary? You can have all the tradeoffs that Java and Python make for quick prototyping if you want to, but then you have a clear path to rock solid code that doesn't involve a complete rewrite.

Reply:
Want to keep the summary short, but I see what you mean. I'll expand it a bit.

Reply:
Expanded!